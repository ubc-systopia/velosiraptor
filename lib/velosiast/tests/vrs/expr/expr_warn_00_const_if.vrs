if true { 4 } else { 2 };
if false { 4 } else { 2 };
if 1 > 0 { 3 } else { 2 };
if 1 < 0 { 3 } else { 2 };