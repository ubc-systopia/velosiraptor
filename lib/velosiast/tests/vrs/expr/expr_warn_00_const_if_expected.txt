0x4
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: The condition of the if-then-else expression is always true.
     --> tests/vrs/expr/expr_warn_00_const_if.vrs:1:4
      |
    1 |  if true { 4 } else { 2 };
      |     ^^^^ This expression is always true.
   ...
      |
    1 |  if true { 4 } else { 2 };
      |            ^ Replace the if-then-else with the expression of the then branch.
warning: generated 1 warnings


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

0x2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: The condition of the if-then-else expression is always false.
     --> tests/vrs/expr/expr_warn_00_const_if.vrs:2:4
      |
    2 |  if false { 4 } else { 2 };
      |     ^^^^^ This expression is always false.
   ...
      |
    2 |  if false { 4 } else { 2 };
      |                        ^ Replace the if-then-else with the expression of the else branch.
warning: generated 1 warnings


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

0x3
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: The condition of the if-then-else expression is always true.
     --> tests/vrs/expr/expr_warn_00_const_if.vrs:3:4
      |
    3 |  if 1 > 0 { 3 } else { 2 };
      |     ^^^^^ This expression is always true.
   ...
      |
    3 |  if 1 > 0 { 3 } else { 2 };
      |             ^ Replace the if-then-else with the expression of the then branch.
warning: generated 1 warnings


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

0x2
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
warning: The condition of the if-then-else expression is always false.
     --> tests/vrs/expr/expr_warn_00_const_if.vrs:4:4
      |
    4 |  if 1 < 0 { 3 } else { 2 };
      |     ^^^^^ This expression is always false.
   ...
      |
    4 |  if 1 < 0 { 3 } else { 2 };
      |                        ^ Replace the if-then-else with the expression of the else branch.
warning: generated 1 warnings


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

